aten/src/ATen/native/cuda/Copy.cu

```diff
@@ -200,23 +200,79 @@ void copy_to_cpu(Tensor& dst, const Tensor& src) {
   });
 }
 
+void copy_from_cpu_async_(Tensor& dst, const Tensor& src) {
+  AT_ASSERT(dst.is_contiguous());
+  AT_ASSERT(src.is_contiguous());
+
+  if (dst.numel() == 0) {
+    return;
+  }
+
+  DeviceGuard device_guard{dst.device()};
+  CUDAStream stream = getCurrentCUDAStream();
+
+  AT_DISPATCH_ALL_TYPES_AND_HALF(src.type(), "copy_from_cpu_async", [&]() {
```

Comment:
I don't think there's any reason to do a dispatch here. You can use void* pointers, and compute the size of the scalar type using itemsize()